src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/mapper/letterdetails/managehearings/ManageHearingFormGLetterDetailsMapper.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.mapper.letterdetails.managehearings;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.finrem.caseorchestration.config.CourtDetailsConfiguration;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.managehearings.Hearing;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.ContactDetailsWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.ManageHearingsWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.document.CourtDetailsTemplateFields;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.letterdetails.DocumentTemplateDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.letterdetails.FormGLetterDetails;
+
+import java.util.Optional;
+
+@Component
+public class ManageHearingFormGLetterDetailsMapper extends AbstractManageHearingsLetterMapper {
+
+    public ManageHearingFormGLetterDetailsMapper(ObjectMapper objectMapper,
+                                                 CourtDetailsConfiguration courtDetailsConfiguration) {
+        super(objectMapper, courtDetailsConfiguration);
+    }
+
+    @Override
+    public DocumentTemplateDetails buildDocumentTemplateDetails(FinremCaseDetails caseDetails) {
+        FinremCaseData caseData = caseDetails.getData();
+
+        Hearing hearing = Optional.ofNullable(caseData.getManageHearingsWrapper())
```

Comment:
You've repeated this code 3 times. It seems like a candidate for putting into the parent class.